### PR TITLE
[CORE-836] Implement `GetBaseline`

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
Pretty straightforward logic change. Mostly adding/updating tests

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/x/ratelimit/keeper/keeper_test.go [hunk 1]
```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
 }
 
 func TestSetGetLimitParams_Success(t *testing.T) {
```
Comment (teddyding) [inline protocol/x/ratelimit/keeper/keeper_test.go:60]: Updated test method to use `tests` instead of a single test

File: protocol/x/ratelimit/keeper/keeper.go [hunk 2]
```diff
@@ -130,8 +127,8 @@ func (k Keeper) GetBaseline(
 	limiter types.Limiter,
 ) *big.Int {
 	// Get the current TVL.
-	// TODO(CORE-836): Query bank Keeper to get current supply of the token.
-	currentTVL := TempTVLPlacerholder
+	supply := k.bankKeeper.GetSupply(ctx, denom)
+	currentTVL := supply.Amount.BigInt()
```
[the remark this answers is not in the document]
Comment (ttl33) [reply protocol/x/ratelimit/keeper/keeper.go:131]: I don't think this comment is relevant since both the current and new CosmosSDK versions won't return errors. In case of internal error, the `GetSupply` returns a 0 coin. 

Of course, panic is a different story

Old: https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fddf421e90574e53af383c1468fdb33d7a3/x/bank/keeper/keeper.go#L218

New: https://github.com/dydxprotocol/cosmos-sdk/blob/b48250d4e0efacbd40f3e0b980c1e1004d29e351/x/bank/keeper/keeper.go#L201

File: protocol/x/ratelimit/keeper/keeper_test.go [hunk 3]
```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
...
+					BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
+					BaselineTvlPpm:  10_000,                         // 1%
+				},
+				{
+					PeriodSec:       86_400,
+					BaselineMinimum: dtypes.NewInt(1_000_000_000_000), // 1m tokens (assuming 6 decimals)
+					BaselineTvlPpm:  100_000,                          // 10%
+				},
+			},
+			expectedCapacityList: []dtypes.SerializableInt{
```
Comment (ttl33) [inline protocol/x/ratelimit/keeper/keeper_test.go:82]: for my own understanding, will there be capacity per denom?

File: protocol/x/ratelimit/keeper/keeper_test.go [hunk 4]
```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
...
+					BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
+					BaselineTvlPpm:  10_000,                         // 1%
+				},
+				{
+					PeriodSec:       86_400,
+					BaselineMinimum: dtypes.NewInt(1_000_000_000_000), // 1m tokens (assuming 6 decimals)
+					BaselineTvlPpm:  100_000,                          // 10%
+				},
+			},
+			balances: []banktypes.Balance{
```
Comment (ttl33) [inline protocol/x/ratelimit/keeper/keeper_test.go:101]: 1. can we add a test case where there's some rounding down or rounding up happens?
2. corner cases: can we add test cases where baseline min and the ppm are 0 or maxmium? 

for example, 10M TVL and the base line ppm are 3% and 30% or something like that?

File: protocol/x/ratelimit/keeper/keeper_test.go [hunk 5]
```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
...
+				},
+			},
+			limiter: types.Limiter{
+				PeriodSec:       3_600,
+				BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k token
+				BaselineTvlPpm:  10_000,                         // 1%
+			},
+			expectedBaseline: big.NewInt(150_000_000_000), // 150k token (1% of 15m)
+		},
+		"max(10% of TVL, $1mm), TVL = 20M token": {
```
Comment (ttl33) [inline protocol/x/ratelimit/keeper/keeper_test.go:266]: why do we use `$1mm` here? is it supposed to be 1m token?

File: protocol/x/ratelimit/keeper/keeper_test.go [hunk 6]
```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
...
+				},
+			},
+			limiter: types.Limiter{
+				PeriodSec:       3_600,
+				BaselineMinimum: dtypes.NewInt(100_000_000_000), // 1m token
+				BaselineTvlPpm:  100_000,                        // 10%
+			},
+			expectedBaseline: big.NewInt(2_000_000_000_000), // 2m token (10% of 20m)
+		},
+		"max(10% of TVL, $1mm), TVL = 8M token": {
```
Comment (ttl33) [inline protocol/x/ratelimit/keeper/keeper_test.go:304]: same here for `$1mm`

File: protocol/x/ratelimit/keeper/keeper_test.go [hunk 7]
```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
...
 
 func TestGetBaseline(t *testing.T) {
-	// TODO(CORE-836): Add test for GetBaseline.
+	tests := map[string]struct {
+		denom            string
+		balances         []banktypes.Balance
+		limiter          types.Limiter
+		expectedBaseline *big.Int
+	}{
+		"max(1% of TVL, 100k token), TVL = 5M token": {
```
Comment (ttl33) [inline protocol/x/ratelimit/keeper/keeper_test.go:199]: similar to above, 
1. can you add test cases that do not divide evenly and would cause rounding down and rounding up?
2. as a corner case, can we add test cases that results in 0 baseline?

**Review by ttl33 — APPROVED** [review]

File: protocol/x/ratelimit/keeper/keeper_test.go (hunk 3 above)
Comment (teddyding) [reply protocol/x/ratelimit/keeper/keeper_test.go:82]: This list is per denom. See test body below - 
```
			// Expected `DenomCapacity` is initialized such that each capacity is equal to the baseline.
			expectedDenomCapacity := types.DenomCapacity{
				Denom:        tc.denom,
				CapacityList: tc.expectedCapacityList,
			}
```

File: protocol/x/ratelimit/keeper/keeper_test.go (hunk 4 above)
Comment (teddyding) [reply protocol/x/ratelimit/keeper/keeper_test.go:101]: 1. Added unit test for rounding.

File: protocol/x/ratelimit/keeper/keeper_test.go (hunk 7 above)
Comment (teddyding) [reply protocol/x/ratelimit/keeper/keeper_test.go:199]: 1. Added such cases
2. As mentioned above, we will not allow `0` baseline or `ppm`
